### PR TITLE
fix(react-components): fix crash related to color overlays

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Bug 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

Fixes out-of-memory issues when color-overlay button is included in UI

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx [hunk 1]
```diff
@@ -26,24 +28,17 @@ export function RuleBasedOutputsSelector({ ruleSet }: ColorOverlayProps): ReactE
...
-
-    currentModel.setDefaultNodeAppearance({
-      color: new Color('#efefef')
-    });
-  });
-
   useEffect(() => {
-    if (!isFetching && hasNextPage !== undefined) {
+    if (!isFetching && hasNextPage === true) {
       void fetchNextPage();
```
Comment (danpriori) [inline react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx:33]: Hmm right. Make sense. So for that we have to change the SearchHooks story as well? Ofc not related to this but assuming that they were using the similar approach.

File: react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /*!
  * Copyright 2023 Cognite AS
  */
-import { useEffect, type ReactElement } from 'react';
+import { useEffect, type ReactElement, useMemo } from 'react';
 
-import { type CogniteCadModel } from '@cognite/reveal';
```
Comment (danpriori) [inline react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx:6]: My Lint is requesting and annoying me to add the type :D

**Review by danpriori — APPROVED** [review]
Some comments but LGTM

File: react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx (hunk 2 above)
Comment (haakonflatval-cognite) [reply react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx:6]: Mine too :sweat_smile: Forgot to run yarn lint

**Pull request merged**